CMakeLists.txt

```diff
@@ -333,12 +333,14 @@ zephyr_cc_option_ifdef(CONFIG_STACK_USAGE            -fstack-usage)
 # in binaries, makes failure logs more deterministic and most
 # importantly makes builds more deterministic
 
-# If both match then the last one wins. This matters for tests/ and
-# samples/ inside *both* CMAKE_SOURCE_DIR and ZEPHYR_BASE: for them
-# let's strip the shortest prefix.
+# If several match then the last one wins. This matters for instances
```

Comment:
Hmm, actually, if this is true, then shouldn't WEST_TOP be matched *before* ZEPHYR_BASE, not after? When using west, ZEPHYR_BASE will usually be inside WEST_TOP, so matching WEST_TOP last will always "win", no? What am I missing?

Reply:
Yes, and WEST_TOP is shorter than ZEPHYR_BASE, which means less substitution and more preserved...?

Reply:
If that's what you want, it seems like the only time when the ZEPHYR_BASE match would win is if the user has something like

```
.
├── random-zephyr-worktree
└── west-top
    └── zephyr
```

E.g. with `west -z random-zephyr-worktree build ...`, you'd see `random-zephyr-worktree/some/path` as `ZEPHYR_BASE/some/path`.

Otherwise (without `-z`) paths inside `west-top/zephyr` would be `WEST_TOP/zephyr/another/path`.

Just checking -- is that really what you mean?

Reply:
Right, the general "policy" of substituting as little as possible does translate (among others) to "use WEST_TOP instead of ZEPHYR_BASE in the most common case". This is my intention.

Reply:
OK, great, that all makes sense, thanks.